runtime/commonMain/src/kotlinx/serialization/internal/GeneratedSerializer.kt

```diff
@@ -10,3 +10,9 @@ import kotlinx.serialization.*
 public interface GeneratedSerializer<T> : KSerializer<T> {
     fun childSerializers(): Array<KSerializer<*>>
 }
+
+@InternalSerializationApi
+@Deprecated("Inserted into generated code and should not be used directly", level = DeprecationLevel.HIDDEN)
+public interface SerializerFactory {
```

Comment:
Please document this annotation so contributors could understand what it is used for